homeassistant/components/nut/diagnostics.py

```diff
@@ -0,0 +1,79 @@
+"""Diagnostics support for Nut."""
+from __future__ import annotations
+
+from typing import Any
+
+from homeassistant.components.diagnostics import async_redact_data
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import device_registry as dr, entity_registry as er
+
+from . import PyNUTData
+from .const import DOMAIN, PYNUT_DATA, PYNUT_UNIQUE_ID
+
+TO_REDACT = {CONF_PASSWORD, CONF_USERNAME}
+
+
+async def async_get_config_entry_diagnostics(
+    hass: HomeAssistant, entry: ConfigEntry
+) -> dict[str, dict[str, Any]]:
+    """Return diagnostics for a config entry."""
+    data = {"entry": async_redact_data(entry.as_dict(), TO_REDACT)}
+    hass_data = hass.data[DOMAIN][entry.entry_id]
+
+    # Get information from Nut library
+    nut_data: PyNUTData = hass_data[PYNUT_DATA]
+    data["nut_data"] = {"ups_list": nut_data.ups_list, "status": nut_data.status}
+
+    # Gather information how this Nut device is represented in Home Assistant
+    device_registry = dr.async_get(hass)
+    entity_registry = er.async_get(hass)
+    hass_device = device_registry.async_get_device(
+        identifiers={(DOMAIN, hass_data[PYNUT_UNIQUE_ID])}
+    )
+    if not hass_device:
+        return data
+
+    data["device"] = {
```

Comment:
Can we use `hass_device.asdict()` here?

Reply:
Does `DeviceEntry` provide `asdict()` method?

Reply:
It uses `attrs` so it should be built in https://www.attrs.org/en/stable/api.html?highlight=asdict#attr.asdict

Reply:
Ok, I used `asdict()` attr method. Do you think that some values should be filtered?